core/storage/trie/serialization/trie_serializer_impl.hpp

```diff
@@ -32,37 +32,45 @@ namespace kagome::storage::trie {
     outcome::result<RootHash> storeTrie(PolkadotTrie &trie,
                                         StateVersion version) override;
 
+    TrieStoreStats const &getLatestStats() const override {
+      return current_stats_;
+    }
+
     outcome::result<std::shared_ptr<PolkadotTrie>> retrieveTrie(
         const common::Buffer &db_key,
         OnNodeLoaded on_node_loaded) const override;
 
-   private:
-    /**
-     * Writes a node to a persistent storage, recursively storing its
-     * descendants as well. Then replaces the node children to dummy nodes to
-     * avoid memory waste
-     */
-    outcome::result<RootHash> storeRootNode(TrieNode &node,
-                                            StateVersion version);
     /**
      * Fetches a node from the storage. A nullptr is returned in case that there
      * is no entry for provided key. Mind that a branch node will have dummy
      * nodes as its children
      */
     outcome::result<PolkadotTrie::NodePtr> retrieveNode(
```

Comment:
Should we use optional instead null-able pointer?

Reply:
I think we should, but I don't want to add this to this PR. I'll leave an issue.